controllers/flinkcluster_util.go

```diff
@@ -533,3 +538,42 @@ func getNonLiveHistory(revisions []*appsv1.ControllerRevision, historyLimit int)
 	nonLiveHistory = append(nonLiveHistory, history[:(historyLen-historyLimit)]...)
 	return nonLiveHistory
 }
+
+func getFlinkJobDeploymentState(flinkJobState string) string {
+	switch flinkJobState {
+	case "INITIALIZING", "CREATED", "RUNNING", "FAILING", "CANCELLING", "RESTARTING", "RECONCILING":
+		return v1beta1.JobStateRunning
+	case "FINISHED":
+		return v1beta1.JobStateSucceeded
+	case "CANCELED":
+		return v1beta1.JobStateCancelled
+	case "FAILED":
+		return v1beta1.JobStateFailed
+	case "SUSPENDED":
+		return v1beta1.JobStateSuspended
+	default:
+		return v1beta1.JobStateUnknown
+	}
+}
+
+// getFlinkJobSubmit extract submit result from the pod termination log.
+func getFlinkJobSubmit(observedPod *corev1.Pod) (*FlinkJobSubmit, error) {
+	if observedPod == nil {
+		return nil, nil
+	}
+	var containerStatuses = observedPod.Status.ContainerStatuses
+	if len(containerStatuses) == 0 ||
+		containerStatuses[0].State.Terminated == nil ||
+		containerStatuses[0].State.Terminated.Message == "" {
+		return nil, nil
+	}
+
+	var rawJobSubmitResult = containerStatuses[0].State.Terminated.Message
```

Comment:
Add a comment here about how the termination message is generated by the submit script. Also add a link to https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/ which could be helpful for some people to understand the mechanism.

Reply:
Commented.